### ISSUE TITLE
How to make changes in a commited branch.

### ISSUE THREAD

**Issue by rverma-jm** [stefanzweifel/git-auto-commit-action#76]
**Version of the Action**
`v4.2.0`

**Describe the bug**
If we run this twice and we have changes in existing target branch, the actions throw error since its expecting that there are other changes in current branch in remote. Possible fix is to allow a -f option to push.

**To Reproduce**
Refer workflow

**Expected behavior**
The idea was to run a periodic job which can run every week and update the timestamp in some files, pushes from master to some branch. If there exists a branch with unmerged changes to master, possibly overwrite those changes forcefully.

**Screenshots**
```
[tf-drift/build]   ⚙  ::set-output:: changes_detected=true
| INPUT_BRANCH value: drift
| error: Your local changes to the following files would be overwritten by checkout:
| 	conf/account-dns/terragrunt.hcl
| 	conf/account-settings/terragrunt.hcl
| 	conf/cdn/terragrunt.hcl
| 	conf/chamber/terragrunt.hcl
| 	conf/cloudtrail/terragrunt.hcl
| 	conf/config/terragrunt.hcl
| 	conf/ecr/terragrunt.hcl
| 	conf/eks/terragrunt.hcl
| 	conf/guard-duty/terragrunt.hcl
| 	conf/iam/terragrunt.hcl
| 	conf/inspector/terragrunt.hcl
| 	conf/kube-platform/terragrunt.hcl
| 	conf/secrets/terragrunt.hcl
| 	conf/service-roles/terragrunt.hcl
| 	conf/tools/terragrunt.hcl
| 	conf/waf/terragrunt.hcl
| Please commit your changes or stash them before you switch branches.
| Aborting
| Error: Invalid status code: 1
|     at ChildProcess.<anonymous> (/actions/stefanzweifel-git-auto-commit-action@v4.2.0/index.js:17:19)
|     at ChildProcess.emit (events.js:311:20)
|     at maybeClose (internal/child_process.js:1021:16)
|     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5) {
|   code: 1
| }
| Error: Invalid status code: 1
|     at ChildProcess.<anonymous> (/actions/stefanzweifel-git-auto-commit-action@v4.2.0/index.js:17:19)
|     at ChildProcess.emit (events.js:311:20)
|     at maybeClose (internal/child_process.js:1021:16)
|     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
[tf-drift/build]   ❌  Failure - stefanzweifel/git-auto-commit-action@v4.2.0
```

**Used Workflow**

```yaml
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
      with:
        ref: master
    - run: |
        git fetch --no-tags --prune --depth=1 origin +refs/heads/*:refs/remotes/origin/*
    - run: for files in $(ls -1A conf/**/terragrunt.hcl); do sed -i '$d' $files && echo $(date '+%d/%m/%Y %H:%M:%S') >> $files ; done;
    - uses: stefanzweifel/git-auto-commit-action@v4.2.0
      with:
        commit_message: Apply automatic terraform drift
        branch: drift
      env:
        GITHUB_TOKEN: ${{ secrets.PAT_TOKEN }}```

**Comment by stefanzweifel** [comment]
Hi,

I'm currently working on a new `push_options`-input in #72 which should work with your use case. You will then be able to add `--force` to your config to allow force pushes.

**Comment by stefanzweifel** [comment]
I've just released [`v4.3.0`](https://github.com/stefanzweifel/git-auto-commit-action/releases/tag/v4.3.0) which now comes with a `push_options`-input.

@rverma-jm  Can you update your workflow like this?

```diff  
-- uses: stefanzweifel/git-auto-commit-action@v4.2.0
+- uses: stefanzweifel/git-auto-commit-action@v4
       with:
         commit_message: Apply automatic terraform drift
         branch: drift
+        push_options: --force
       env:
         GITHUB_TOKEN: ${{ secrets.PAT_TOKEN }}
```

The switch to `v4` will keep you always up to date on the latest release. `push_options: --force` will force push the commit to the repository.

**Comment by stefanzweifel** [comment]
@rverma-jm I'm closing this issue now, as it seems resolved for me.
Feel free to reopen this issue or create a new one, if the problem persists for you.
